File: src/main/go/array/remove_duplicates_from_sorted_array.go
package array

func removeDuplicates(nums []int) int {
	if len(nums) == 0 {
		return 0
	}

	j := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] != nums[j] {
			i++
			nums[i] = nums[j]
		}
	}

	return j + 1

}

File: src/main/java/leetcode/array/RemoveDuplicatesfromSortedArray.java
package leetcode.array;

import java.util.*;

public class RemoveDuplicatesfromSortedArray {
  public int removeDuplicates(int[] nums) {
    HashSet<Integer> set = new HashSet<>();
    int i = 0;
    for (int j = 0; j < nums.length; j++) {
      if (!set.contains(nums[j])) {
        nums[i++] = nums[j];
        set.add(nums[j]);
      }
    }


    return set.size();
  }
}
